enumerables.rb

```diff
@@ -0,0 +1,66 @@
+# enumerable.rb
+module Enumerable
+  def my_each
+    i = 0
+    while to_a.length > i
+      yield(to_a[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while to_a.length > i
+      yield(to_a[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    new_array = []
+    to_a.my_each { |item| new_array.push(item) if yield(item) }
+    new_array
+  end
+
+  def my_all?
+    to_a.my_each { |item| return false if yield(item) == false }
+    true
+  end
+
+  def my_any?
+    to_a.my_each { |item| return true if yield(item) }
+    false
+  end
+
+  def my_none?
+    to_a.my_each { |item| return false if yield(item) }
+    true
+  end
+
+  def my_count?
+    i = 0
+    to_a.my_each { |_item| i += 1 }
+    i
+  end
```

Comment:
- [ ] enumerables #my_count returns the number of items in enum through enumeration
      Failure/Error: expect(array.my_count).to eq array.count

- [ ] enumerables #my_count returns the number of items in enum through enumeration::range
      Failure/Error: expect(range.my_count).to eq range.count

- [ ] enumerables #my_count counts the number of items in enum that are equal to item if an argument is given
      Failure/Error: expect(array.my_count(LOWEST_VALUE)).to eq array.count(LOWEST_VALUE)

- [ ] enumerables #my_count counts the number of elements yielding a true value if a block is given
      Failure/Error: expect(array.my_count(&block)).to eq array.count(&block)

- [ ] enumerables #my_count does not mutate the original array
      Failure/Error: array.my_count { |num| num + 1 }

Reply:
Solved all the requested changes and also tested it